src/psyclone/tests/psyir/transformations/extract_test.py

```diff
@@ -129,3 +133,106 @@ def test_malformed_extract_node(monkeypatch):
     with pytest.raises(InternalError) as err:
         _ = enode.extract_body
     assert "malformed or incomplete. It should have a " in str(err.value)
+
+
+# --------------------------------------------------------------------------- #
+def test_get_default_options():
+    '''Check the default options.'''
+
+    etrans = ExtractTrans()
+    assert etrans.get_default_options() == {"COLLECT-ARRAY-SHAPE-READS": True}
+
+
+# -----------------------------------------------------------------------------
+def test_extract_validate():
+    '''Test that the validate function can successfully finish.'''
+
+    _, invoke = get_invoke("single_invoke_three_kernels.f90",
+                           "gocean1.0", idx=0, dist_mem=False)
+    etrans = ExtractTrans()
+    etrans.apply(invoke.schedule.children[0])
```

Comment:
Better to just call `validate` directly?

Reply:
Indeed.